### ISSUE TITLE
Deploy .pb frozen model in Android mobile

### ISSUE THREAD

**Issue by abdou31** [tensorflow/tensorflow#30064]
Hello,
I have worked on iris detection ( in the human eye ) in the real-time project using deep learning. So I have used a CNN model based on facial landmarks detection using Tensorflow GPU version with regression predict and I have modified the number of points to get only the iris region, 
I have now .ckpt files trained the model with 146 000 steps and I have tried to freeze the model but I'm not sure that the process of the freeze is correct or no.
I get a .pb file with size 28 Mo.
I want to deploy and use this .pb file into an Android application to detect the iris region by drawing the landmarks on it.
How can I use a .pb file for Android application?
How can I extract the prediction from .pb file in the Android application?
Thanks

**Comment by gadagashwini** [comment]
@abdou31 This [document](https://www.tensorflow.org/lite/convert/cmdline_examples) will guide you how to convert frozen graph to lite model. Let us know if that helps you. Thanks!

**Comment by abdou31** [comment]
Thanks, I will see it

**Comment by gadagashwini** [comment]
@abdou31 Can we close this issue.Since it is not bug/performance or build/install issue. Thanks!

**Comment by abdou31** [comment]
I have successfully converted .pb frozen model to tflite model but the problem I didn't know that this is correct or no.
Do you have idea how can I know that the file generated ( tflite ) is corrected?
I write this because I didn't have any about output and input nodes setted are correct or no?

**Comment by gadagashwini** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by abdou31** [comment]
Ok, but can you tell me where I can post this issue?

**Comment by gadagashwini** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow). Thanks!

**Comment by gadagashwini** [comment]
@abdou31 Have look on [tensorflow lite guide](https://www.tensorflow.org/lite/guide/android) for android app. Thanks!

**Comment by abdou31** [comment]
Actually, I ask this question in StackOverflow but I didn't get any answer.
I don't know why but I think that the community didn't give any importance to questions related to IA (Deep learning  and Machine learning).

**Comment by ymodak** [comment]
Closing this issue since the original title of the issue is resolved. Please post a new issue if facing additional problems unrelated to original issue. This will help us to keep track of the topics discussed. Thanks!

**Comment by abdou31** [comment]
This problem is solved.
To deploy a model into Android Studio. it should using a tflite model not a pb model

**Comment by shbnm21** [comment]
@abdou31 have u able to deploy the model into Android?
If yes can u tell me how did u test  tflite model?

**Comment by gadagashwini** [comment]
@shbnm21, 
Please take a look at [android guide doc](https://www.tensorflow.org/lite/guide/android).

**Comment by shbnm21** [comment]
@gadagashwini I will look into that.
I have used a pretrained model (GAN model) for face attribute manipulation .converted into .pb file.The size of .pb file is 166mb.The model is based on tensorflow 1.15 .I have googled about deployment of the model on Android.some blogs says to use .pb file into Android and some says to use tflite .
Now I am confused can u provide any documentation on how to convert .pb file into tflite and to test it's accuracy.
Also I want to know whether tflite can be used to convert GAN models.

**Comment by gadagashwini** [comment]
@shbnm21,
Please find the documentation on how to convert pb to lite model in this [link](https://www.tensorflow.org/lite/convert/cmdline_examples#convert_a_tensorflow_graphdef_). Thanks!

**Comment by shbnm21** [comment]
@gadagashwini thank you

**Comment by nihkilft123** [comment]
You are right there are some apps [like this](https://play.google.com/store/apps/details?id=com.bgstudio.auto.removebg) which using direct .pb file in their app. I don't know how but I found they are using .pb (Frozen model) in their app! So, its possible there is some technique must be to use it!

**Comment by shbnm21** [comment]
@nihkilft123 
Do u know anything how faceapp has deployed their model on android?
My understanding is they have created API and process the image on server and send back the results on Android.
